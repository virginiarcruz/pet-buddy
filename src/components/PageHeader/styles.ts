import styled from 'styled-components'
import media from 'styled-media-query'

export const Header = styled.header`
  margin-bottom: 32px;
  > input {
    margin-top: 32px;
  }
  ${media.greaterThan('medium')`
    display: flex;
    justify-content: space-between;
    align-items: center;
    margin-bottom: 0;
    > input {
      margin-top: 0;
    }
  `}
`
