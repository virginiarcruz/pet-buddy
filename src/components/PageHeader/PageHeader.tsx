import React from 'react'
import Heading from 'components/Heading'
import Input from 'components/Input'

import { Header } from './styles'

const PageHeader = () => (
  <Header>
    <Heading title="Your Animals" labelBadge="28" />
    <Input placeholder="Search an animal by name" />
  </Header>
)

export default PageHeader
