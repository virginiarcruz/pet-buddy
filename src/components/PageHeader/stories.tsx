import { Story, Meta } from '@storybook/react/types-6-0'
import PageHeader from '.'

export default {
  title: 'PageHeader',
  component: PageHeader
} as Meta

export const Default: Story = () => <PageHeader />
