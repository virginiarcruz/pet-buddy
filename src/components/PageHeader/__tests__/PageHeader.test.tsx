import React from 'react'
import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/tests/helpers'

import PageHeader from '../PageHeader'

describe('<PageHeader />', () => {
  it('should render the <PageHeader />', () => {
    const { container } = renderWithTheme(<PageHeader />)

    expect(container.firstChild).toMatchSnapshot()
  })
})
