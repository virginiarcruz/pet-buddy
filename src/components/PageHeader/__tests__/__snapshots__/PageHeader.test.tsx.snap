// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PageHeader /> should render the <PageHeader /> 1`] = `
.c3 {
  border-radius: 100px;
  background-color: #F47500;
  color: #FFFFFF;
  font-weight: 600;
  font-size: 14px;
  padding: 4px 8px;
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
  height: 25px;
  line-height: 17px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c2 {
  color: #000121;
  font-family: 'Inter',-apple-system,sans-serif;
  font-size: 22px;
  font-weight: 600;
  margin-right: 8px;
  line-height: 27px;
}

.c4 {
  padding: 12px 12px 12px 32px;
  background-color: #F4F4F7;
  background-image: url('img/icon-search.svg');
  background-repeat: no-repeat;
  background-position: 12px 14px;
  border-radius: 12px;
  border: 0;
  font-size: 16px;
  font-family: 'Inter',-apple-system,sans-serif;
  color: #A1A2AF;
  line-height: 19px;
  width: 100%;
}

.c4::-webkit-input-placeholder {
  color: #A1A2AF;
}

.c4::-moz-placeholder {
  color: #A1A2AF;
}

.c4:-ms-input-placeholder {
  color: #A1A2AF;
}

.c4::placeholder {
  color: #A1A2AF;
}

.c0 {
  margin-bottom: 32px;
}

.c0 > input {
  margin-top: 32px;
}

@media (min-width:992px) {
  .c4 {
    width: 310px;
  }
}

@media (min-width:992px) {
  .c0 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-box-pack: justify;
    -webkit-justify-content: space-between;
    -ms-flex-pack: justify;
    justify-content: space-between;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    margin-bottom: 0;
  }

  .c0 > input {
    margin-top: 0;
  }
}

<header
  class="c0"
>
  <div
    class="c1"
  >
    <h1
      class="c2"
    >
      Your Animals
    </h1>
    <span
      class="c3"
    />
  </div>
  <input
    class="c4"
    id="inputFilter"
    placeholder="Search an animal by name"
    type="text"
    value=""
  />
</header>
`;
